win32/devdoc/callonce_requirements.md

```diff
@@ -0,0 +1,87 @@
+# call_once requirements
+================
+
+## Overview
+
+call_once is a module that provides a pair of functions callonce_begin/callonce_end to be used to have some code executed just once. 
+
+Typical usage is:
+
+```c
+
+volatile_atomic int32_t wasInit = CALL_ONCE_NOT_CALLED;

```

Comment:
>volatile_atomic int32_t [](start = 0, length = 23)

Maybe this should be a typedef just because it's a "special" volatile int that the caller shouldn't really do anything with besides initialize.

Maybe even better to make a macro to hide "CALL_ONCE_NOT_CALLED" and have something like DECLARE_LAZY_FLAG(name)....

Reply:
I will change it to typedef.

---
In reply to: [466039465](https://github.com/Azure/azure-c-pal/pull/28#discussion_r466039465) [](ancestors = 466039465)

Reply:
changed to call_once_t

---
In reply to: [466125526](https://github.com/Azure/azure-c-pal/pull/28#discussion_r466125526) [](ancestors = 466125526,466039465)